task-launcher/src/tasks/shared/helpers/audioHandler.ts

```diff
@@ -9,6 +9,13 @@ export class PageAudioHandler {
   static audioContext: BaseAudioContext;
   static audioUri: string;
   static audioSource?: AudioBufferSourceNode;
+  static replayPresses: number;
+
+  static saveReplayPresses(){
+    jsPsych.data.addDataToLastTrial({
+      audioButtonPresses: PageAudioHandler.replayPresses
```

Comment:
Hmm, so we are making the `PageAudioHandler` aware to the data saving to the backend? I am fine with it, usually I like to keep these classes stateless. In this case that would mean instead of calling `saveReplayButton()`, I would just do: `jsPsych.data.addDataToLastTrial({ audioButtonPresses: PageAudioHandler.replayPresses})` .

But I won't be very fussy about this :). I will leave it up to you. I am fine either way.

Reply:
That sounds good, thanks! I was going back and forth about which way would make more sense.